shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -171,7 +163,20 @@ - (BOOL)application:(UIApplication*)application
   if ([_lifeCycleDelegate application:application openURL:url options:options]) {
     return YES;
   }
-  return [self openURL:url];
+
+  __block BOOL openURLSuccess = NO;
+  __block BOOL openURLCompleted = NO;
+  [self openURL:url
+                options:options
+      completionHandler:^(BOOL success) {
+        openURLSuccess = success;
+        openURLCompleted = YES;
+      }];
+
+  while (!openURLCompleted) {
+    [NSRunLoop.currentRunLoop runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.1]];
+  }
```

Comment:
Optional: If you are iffy about using BOOL flag for completeness, a semaphore version can be (pseudocode): 

```
__block BOOL openURLSuccess = NO;
dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
[self openURL: url ... {
    openURLSuccess = success;
    dispatch_semaphore_signal(semaphore)
}];

while (dispatch_semaphore_wait(semaphore, DISPATCH_TIME_NOW)) {
    [NSRunLoop.currentRunLoop runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.1]];
}
```
Though a BOOL flag would be equally as good, since callback happens on main thread. I don't have a preference here.

Reply:
Oh I just saw @jmagman's [comment on semaphore](https://github.com/flutter/engine/pull/52643/files/77c56fe9519815334239f0ea82ab472d8a6cd664#r1600612505). 

To use semaphore on the same thread, you can't block it due to deadlock, as jenn previously commented. Instead, you pass in `DISPATCH_TIME_NOW`, and kick off the runloop so it can pick up the completion handler.